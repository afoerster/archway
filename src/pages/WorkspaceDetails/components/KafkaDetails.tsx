import { Card, Row } from 'antd';
import * as React from 'react';
import { KafkaTopic } from '../../../types/Workspace';
import CardHeader from './CardHeader';

interface Props {
    consumerGroup: string;
    topics: KafkaTopic[];

    showModal: (e: React.MouseEvent) => void;
}

const renderTopic = (topic: KafkaTopic) => (
  <div>topic.name</div>
);

const KafkaDetails = ({ consumerGroup, topics, showModal }: Props) => (
  <Card
    actions={[
      <a href="#" onClick={showModal}>Add a topic</a>,
    ]}>
    <CardHeader
      icon="sound"
      heading="Kafka Topics"
      subheading={consumerGroup} />
    <Row gutter={12} type="flex" justify="center" style={{ marginTop: 18 }}>
      {topics && topics.length > 0 && topics.map(renderTopic)}
      {(!topics || topics.length <= 0) && (
        <div style={{ color: 'rgba(0, 0, 0, .65)' }}>
          No topics yet.
        </div>
      )}
    </Row>
  </Card>
);

export default KafkaDetails;
