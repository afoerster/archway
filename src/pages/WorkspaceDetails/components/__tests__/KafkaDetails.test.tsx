import * as React from 'react';
import { shallow } from 'enzyme';

import KafkaDetails from '../KafkaDetails';

describe('KafkaDetails', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<KafkaDetails consumerGroup="" topics={[]} showModal={() => null} />);
    expect(wrapper).toMatchSnapshot();
  });
});
