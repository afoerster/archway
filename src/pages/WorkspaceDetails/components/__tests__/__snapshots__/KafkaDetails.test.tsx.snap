// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KafkaDetails renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <KafkaDetails
    consumerGroup=""
    showModal={[Function]}
    topics={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "actions": Array [
        <a
          href="#"
          onClick={[Function]}
        >
          Add a topic
        </a>,
      ],
      "children": Array [
        <div
          style={
            Object {
              "display": "flex",
              "justifyContent": "space-between",
            }
          }
        >
          <Label
            style={
              Object {
                "lineHeight": "20px",
              }
            }
          >
            <Icon
              style={
                Object {
                  "fontSize": 20,
                  "paddingRight": 5,
                }
              }
              theme="twoTone"
              twoToneColor="rgb(67, 170, 139)"
              type="sound"
            />
            Kafka
          </Label>
          <Label
            style={
              Object {
                "fontSize": 10,
                "lineHeight": "18px",
              }
            }
          >
            
          </Label>
        </div>,
        <List
          bordered={false}
          dataSource={Array []}
          loading={false}
          locale={
            Object {
              "emptyText": "No topics yet",
            }
          }
          pagination={false}
          prefixCls="ant-list"
          renderItem={[Function]}
          split={true}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Label
              style={
                Object {
                  "lineHeight": "20px",
                }
              }
            >
              <Icon
                style={
                  Object {
                    "fontSize": 20,
                    "paddingRight": 5,
                  }
                }
                theme="twoTone"
                twoToneColor="rgb(67, 170, 139)"
                type="sound"
              />
              Kafka
            </Label>,
            <Label
              style={
                Object {
                  "fontSize": 10,
                  "lineHeight": "18px",
                }
              }
            >
              
            </Label>,
          ],
          "style": Object {
            "display": "flex",
            "justifyContent": "space-between",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <Icon
                  style={
                    Object {
                      "fontSize": 20,
                      "paddingRight": 5,
                    }
                  }
                  theme="twoTone"
                  twoToneColor="rgb(67, 170, 139)"
                  type="sound"
                />,
                "Kafka",
              ],
              "style": Object {
                "lineHeight": "20px",
              },
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "style": Object {
                    "fontSize": 20,
                    "paddingRight": 5,
                  },
                  "theme": "twoTone",
                  "twoToneColor": "rgb(67, 170, 139)",
                  "type": "sound",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "Kafka",
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "",
              "style": Object {
                "fontSize": 10,
                "lineHeight": "18px",
              },
            },
            "ref": null,
            "rendered": "",
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bordered": false,
          "dataSource": Array [],
          "loading": false,
          "locale": Object {
            "emptyText": "No topics yet",
          },
          "pagination": false,
          "prefixCls": "ant-list",
          "renderItem": [Function],
          "split": true,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "actions": Array [
          <a
            href="#"
            onClick={[Function]}
          >
            Add a topic
          </a>,
        ],
        "children": Array [
          <div
            style={
              Object {
                "display": "flex",
                "justifyContent": "space-between",
              }
            }
          >
            <Label
              style={
                Object {
                  "lineHeight": "20px",
                }
              }
            >
              <Icon
                style={
                  Object {
                    "fontSize": 20,
                    "paddingRight": 5,
                  }
                }
                theme="twoTone"
                twoToneColor="rgb(67, 170, 139)"
                type="sound"
              />
              Kafka
            </Label>
            <Label
              style={
                Object {
                  "fontSize": 10,
                  "lineHeight": "18px",
                }
              }
            >
              
            </Label>
          </div>,
          <List
            bordered={false}
            dataSource={Array []}
            loading={false}
            locale={
              Object {
                "emptyText": "No topics yet",
              }
            }
            pagination={false}
            prefixCls="ant-list"
            renderItem={[Function]}
            split={true}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Label
                style={
                  Object {
                    "lineHeight": "20px",
                  }
                }
              >
                <Icon
                  style={
                    Object {
                      "fontSize": 20,
                      "paddingRight": 5,
                    }
                  }
                  theme="twoTone"
                  twoToneColor="rgb(67, 170, 139)"
                  type="sound"
                />
                Kafka
              </Label>,
              <Label
                style={
                  Object {
                    "fontSize": 10,
                    "lineHeight": "18px",
                  }
                }
              >
                
              </Label>,
            ],
            "style": Object {
              "display": "flex",
              "justifyContent": "space-between",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <Icon
                    style={
                      Object {
                        "fontSize": 20,
                        "paddingRight": 5,
                      }
                    }
                    theme="twoTone"
                    twoToneColor="rgb(67, 170, 139)"
                    type="sound"
                  />,
                  "Kafka",
                ],
                "style": Object {
                  "lineHeight": "20px",
                },
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "style": Object {
                      "fontSize": 20,
                      "paddingRight": 5,
                    },
                    "theme": "twoTone",
                    "twoToneColor": "rgb(67, 170, 139)",
                    "type": "sound",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "Kafka",
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "",
                "style": Object {
                  "fontSize": 10,
                  "lineHeight": "18px",
                },
              },
              "ref": null,
              "rendered": "",
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bordered": false,
            "dataSource": Array [],
            "loading": false,
            "locale": Object {
              "emptyText": "No topics yet",
            },
            "pagination": false,
            "prefixCls": "ant-list",
            "renderItem": [Function],
            "split": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
